### PR TITLE
Improve Docs

### PR DESCRIPTION
**Pull request by boojamya**
This PR:

- updates the home Readme
- Improves documentation for running conformance tests
- Adds documentation on how to construct custom tests
   - It allows the learner to follow along with a basic testing file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,6 @@ ibctest.test
 /bin
 
 #GoLand project directory
-.idea/
\ No newline at end of file
+.idea/
+
+.vscode/
```
Comment (DavidNix) [inline .gitignore:10]: `.idea/` was likely my fault. 

These types of files are ideally in your [global gitignore](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c), so they aren't committed to the repo. So, best practices is delete `.idea` and `.vscode` here and instead add them to your global gitignore.

File: docs/architectCustomTests.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+        },
+        Bin: "stringd",
+        Bech32Prefix: "cosmos",
+        Denom: "cheese",
+        GasPrices: "0.01cheese",
+        GasAdjustment: 1.3,
+        TrustingPeriod: "508h",
+        NoHostMount: false},
+    },
+    })
```
Comment (DavidNix) [inline docs/architectCustomTests.md:84]: I appreciate all the code samples in the markdown here. Unfortunately, they risk becoming stale and inaccurate quickly. And there's nothing besides developer discipline reminding us to update this document.

I recommend moving this document's content as comments in `examples/learn_ibc_test.go`. Because `learn_ibc_test.go` is actual code, we'll be forced to keep it up to date as ibctest gets extended and refactored. Then this document becomes sparse, perhaps laying out some high level detail and informing the user to check out the example test.

File: docs/ciTests.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Integrate Tests Into Github Client
+
+### Example implementatios:
+- Go Relayer - https://github.com/cosmos/relayer/blob/main/.github/workflows/ibctest.yml
+- IBC-Go e2e tests - https://github.com/cosmos/ibc-go/blob/main/.github/workflows/e2e-test-workflow-call.yml
```
Comment (DavidNix) [inline docs/ciTests.md:5]: 💯 - reference what's actually working!

File: README.md [hunk 4]
```diff
@@ -1,68 +1,57 @@
...
-ibctest -test.run=/////height_timeout
-ibctest -test.run=/////timestamp_timeout
-```
+## Table Of Contents
+- [Building Binary](#building-binary)
+- **Usage:**
+    - [Running Conformance Tests](./docs/conformanceTests.md) - suite of built-in tests that test high-level IBC compatibility
+    - [Architect Custom Tests](./docs/architectCustomTests.md) - How to create custom tests
+- [Retaining Data on Failed Tests](./docs/retainingDataOnFailedTests.md)
+- [Deploy as GitHub CI Tests](./docs/ciTests.md)
```
Comment (DavidNix) [inline README.md:30]: I ❤️ this pattern. The primary README is an index to smaller, focused documentation.

File: README.md [hunk 5]
```diff
@@ -1,68 +1,57 @@
...
-ibctest -test.run=/////relay_packet
-ibctest -test.run=/////no_timeout
-ibctest -test.run=/////height_timeout
-ibctest -test.run=/////timestamp_timeout
-```
+## Table Of Contents
+- [Building Binary](#building-binary)
+- **Usage:**
+    - [Running Conformance Tests](./docs/conformanceTests.md) - suite of built-in tests that test high-level IBC compatibility
+    - [Architect Custom Tests](./docs/architectCustomTests.md) - How to create custom tests
```
Comment (DavidNix) [inline README.md:28]: nit:

`Architect Custom Tests` -> `Write Custom Tests`?

**Review by DavidNix — COMMENTED** [review]
Great work! My one suggestion of moving documentation into comments of `learn_ibc_test.go` is my one blocker.

File: .gitignore (hunk 1 above)
Comment (boojamya) [reply .gitignore]: Oh Nice!

File: README.md (hunk 5 above)
Comment (boojamya) [reply README.md]: Yea... that probably is the better word to use. Done!

File: docs/architectCustomTests.md (hunk 2 above)
Comment (boojamya) [reply docs/architectCustomTests.md]: I partially disagree here. 
Since this is a doc on how to write custom tests, I think the code snippets help break up the document making it easier to read and give context to what a "Chain Factory", "Interchain", ect. look like.
As for the code getting stale... I really did try and keep this high level. But maybe I am taking for granted the rate of refactors!?

Maybe there are too many code snippets? There's a few extras in here just to show the learner that there are multiple ways to do the same thing, some of which are 1) hard to discover and 2) important for the writer to know depending on what they are testing.

For me personally, this Is how I learn best — having a script that is complete, and then having a description of each section. I think it would be a much tougher read if this was commented out in a go file. Given the audience, perhaps this is too wordy and unnecessary?

I removed some code and condensed a little in the latest push. 
If consensus is to make this less code and more high level explanations, I’m happy to make more adjustments. Regardless, appreciate any/all feedback.

**Comment by boojamya** [comment]
Forgot to add note about sqlite db...

File: docs/architectCustomTests.md (hunk 2 above)
Comment (DavidNix) [reply docs/architectCustomTests.md]: Go also has a feature for [testable examples](https://go.dev/blog/examples) we may be able to leverage. I'm not sure if that's a good fit there, but it's another option - perhaps for a future PR.

**Review by DavidNix — APPROVED** [review]

**Comment by jackzampolin** [comment]
We can add an issue to do the docs for the sqlite feature
